gate-web/src/main/groovy/com/netflix/spinnaker/gate/interceptors/RequestSheddingInterceptor.java

```diff
@@ -0,0 +1,161 @@
+ * request header, which must also include a "low" value. If missing
+ * or given any other value, the request will be treated as as normal.
+ *
+ * Uses the dynamic config service to enable/disable the functionality,
+ * and further to optionally drop low priority requests on specific
+ * request paths. If no request paths are set, then all low priority
+ * requests will be dropped.
+ *
+ * A percentage chance config key is also available which takes a
+ * value between 0 and 100, representing the percentage chance that
+ * the request will be dropped. If the value is set to "60", there
+ * will be a 60% chance the request will be dropped; this value is
+ * only considered when the enabled key is also set to "true".
+ *
+ * If enabled but an internal error occurs due to configuration, the
+ * interceptor will elect to drop all low priority requests.
+ */
+public class RequestSheddingInterceptor extends HandlerInterceptorAdapter {
+
+  final static String PRIORITY_HEADER = "X-Spinnaker-Priority";
+  final static String LOW_PRIORITY = "low";
+
+  final static String ENABLED_KEY = "requestShedding";
+  final static String CHANCE_KEY = "requestShedding.chance";
+  final static String PATHS_KEY = "requestShedding.paths";
+
+  private final static Logger log = LoggerFactory.getLogger(RequestSheddingInterceptor.class);
+  private final static Splitter splitter = Splitter.on(",").omitEmptyStrings().trimResults();
+  private final static Random random = new Random();
+
+  private final DynamicConfigService configService;
+  private final Registry registry;
+
+  private final Id requestsId;
+
+  public RequestSheddingInterceptor(DynamicConfigService configService, Registry registry) {
+    this.configService = configService;
+    this.registry = registry;
+
+    this.requestsId = registry.createId("requestShedding.requests");
+  }
+
+  @Override
+  public boolean preHandle(HttpServletRequest request,
+                           HttpServletResponse response,
+                           Object handler) throws Exception {
+    if (configService.isEnabled(ENABLED_KEY, false) && isDroppable(request)) {
+      if (shouldDropRequestWithChance()) {
+        log.warn("Dropping low priority request: {}", request.getRequestURI());
+        registry.counter(requestsId.withTag("action", "dropped")).increment();
+
+        response.setDateHeader("Retry-After", Instant.now().plus(Duration.ofMinutes(1)).toEpochMilli());
+        throw new LowPriorityRequestRejected();
+      }
+      log.debug("Dice roll prevented low priority request shedding: {}", request.getRequestURI());
+      registry.counter(requestsId.withTag("action", "allowed")).increment();
+    }
+
+    return super.preHandle(request, response, handler);
+  }
+
+  private boolean isDroppable(HttpServletRequest request) {
+    String uri = request.getRequestURI();
+
+    // We want our error handler to service the errors correctly
+    if ("/error".equals(uri)) {
```

Comment:
what about `/healthcheck` requests? Are they handled by a something different?

Reply:
They seem to be, yes. I only needed to make an exception for `/error`.

Reply:
isn't that what `/*` is for?